tests/e2e/policy_driven_vol_allocation.go

```diff
@@ -1374,6 +1377,635 @@ var _ = ginkgo.Describe("[vol-allocation] Policy driven volume space allocation
+		n := resource.MustParse(largeSize)
+		sizeInInt, b := n.AsInt64()
+		gomega.Expect(b).To(gomega.BeTrue())
+		// since we are creating 4 volumes reducing the size of each expansion by half to start with
+		newSize := *(resource.NewQuantity(sizeInInt/2, resource.BinarySI))
+		start := time.Now()
+		for i, volId := range volIds {
+			go reconfigPolicyParallel(ctx, volId, policyIds[i/2].UniqueId, &wg)
+		}
+		wg.Wait()
+
+		wg.Add(len(volIds))
+		for _, pvc := range pvcs {
+			go resize(client, pvc, pvc.Spec.Resources.Requests[v1.ResourceStorage], newSize, &wg)
+		}
+		wg.Wait()
+		for i := range pvcs {
+			err = waitForPvResize(pvs[i], client, newSize, totalResizeWaitPeriod*2)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred(), "resize exceeded timeout")
+		}
+		elapsed := time.Since(start)
+
+		ginkgo.By("Verify PVC expansion took longer than vpxd timeout")
+		gomega.Expect(elapsed > time.Second*time.Duration(vpxdReducedTaskTimeoutSecsInt)).To(
+			gomega.BeTrue(), "PVC expansion was faster than vpxd timeout")
+
+		ginkgo.By("Wait and verify the file system resize on pvcs")
+		for i := range pods {
+			framework.Logf("Waiting for file system resize to finish for pvc %v", pvcs[i*2].Name)
+			pvcs[i*2], err = waitForFSResize(pvcs[i*2], client)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			fsSize, err := getFSSizeMb(f, pods[i])
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			framework.Logf("File system size after expansion : %v", fsSize)
+			gomega.Expect(fsSize > fsSizes[i]).To(gomega.BeTrue(),
+				fmt.Sprintf(
+					"filesystem size %v is not > than before expansion %v for pvc %q",
+					fsSize, fsSizes[i], pvcs[i*2].Name))
+
+			framework.Logf("File system resize finished successfully for pvc %v", pvcs[i*2].Name)
+		}
+
+	})
+
+	/*
+		relocate from vmfs datastore to another
+		Steps for offline volumes:
+		1	Create a vmfs SPBM policy with thin volume allocation
+		2	Create a SC using policy created in step 1
+		3	Create a PVC using SC created in step 2
+		4	Verify that pvc created in step 3 is bound
+		5	Create a pod, say pod1 using pvc created in step 4.
+		6	write some data to the volume.
+		7	delete pod1.
+		8	Relocate volume to another vmfs datastore.
+		9	Recreate pod1
+		10	Verify pod1 is running and pvc1 is accessible and verify the data written in step 6
+		11	Delete pod1
+		12	Delete pvc, sc and SPBM policy created for this test
+
+		Steps for online volumes:
+		1	Create a vmfs SPBM policies with thin volume allocation
+		2	Create a SC using policy created in step 1
+		3	Create a PVC using SC created in step 2
+		4	Verify that pvc created in step 3 is bound
+		5	Create a pod, say pod1 using pvc created in step 4.
+		6	write some data to the volume.
+		7	Relocate volume to another vmfs datastore.
+		8	Verify the data written in step 6
+		9	Delete pod1
+		10	Delete pvc, sc and SPBM policy created for this test
+	*/
+	ginkgo.It("[csi-block-vanilla] relocate from one vmfs datastore to another", func() {
+
```

Comment:
This testcase should be relocating from vmfs to vsan datastore right instead from vmfs to vmfs? We already have a testcase where we are relocating from 1 vmfs ds to another vmfs ds.

Reply:
I wanted to fix the testcase, I have messed up my code merge, one more test is missing, my bad!

Reply:
Updated the exiting testcase with the changes.